azuredevops/internal/service/git/resource_git_repository_branch.go

```diff
@@ -0,0 +1,239 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/git"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceGitRepositoryBranch schema to manage the lifecycle of a git repository branch
+func ResourceGitRepositoryBranch() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGitRepositoryBranchCreate,
+		ReadContext:   resourceGitRepositoryBranchRead,
+		DeleteContext: resourceGitRepositoryBranchDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGitRepositoryBranchImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"repository_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			"ref": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"tag": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"commit_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"branch_reference": {
```

Comment:
`branch_reference` is bit duplicate of `name` , the only difference is the prefix

Reply:
I think name should be without prefix, because GetBranch api expects name without prefix and in the GUI pages it is also without prefix.

But other resources in the azure devops provider like git_repository_file, git_repository, branch policies expect branch name with prefix. This `branch_reference` field can be used to pass valid branch name to those.

 Some options are:
- Force name to have prefix
- Remove this field and keep name the same without prefix
- Change git_repository_file and other resources to use branch name without prefix

My preference is the first or the second one. The first one is  more like branch names in other resources.  In the second one we can let users add prefix like `"refs/heads/${azuredevops_git_repository_branch.example.name}"`.

Reply:
Can also remove `branch_reference` and allow `name` both with and without prefix.

Reply:
For now I went with option 2. But can easily change to option 1. Curious what others think.

Reply:
I prefer the second one, let the user decide which prefix they want.